lib/embroider/update-manifest-from-html.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const fs = require('fs');
+const findScriptSrcs = require('find-scripts-srcs-in-document');
+const isEmbroider = require('./is-embroider');
+
+// TODO: hard coding these seem unfortunate, especially vendor-static which is non-standard
+const FILES_TO_SKIP = /vendor|vendor-static/gi;
+
+module.exports = function updateManifestFromHtml(root) {
+  const pkg = JSON.parse(fs.readFileSync(root + '/package.json', 'UTF8'));
+
+  // bail if not embroider (let's chat if this is appropriate);
+  if (isEmbroider(root)) {
+    return;
+  }
+
+  // otherwise, we must parse the index.html file to figure out what's next
+  const indexHtml = fs.readFileSync(root + '/index.html', 'UTF8')
+  // TODO: fix simple-html-tokenizer: https://github.com/tildeio/simple-html-tokenizer/pull/71
+    .replace('<!DOCTYPE html>','');
+
+  const indexScriptFilePaths = findScriptSrcs(indexHtml, findScriptSrcs.ignoreWithAttribute('data-embroider-ignore'))
+    // TODO: these FILES_TO_SKIP things seem over zealous
+    .filter( src => !FILES_TO_SKIP.test(src));
+
+    const appFiles = [];
+    indexScriptFilePaths.forEach(src => {
+      // Extract the file name from the src file paths and
+      // checks the file exits in the assets location on the disk.
+      const scriptFileNameRegEx = /([a-zA-Z0-9_\.\-\(\):])+(\.js)/ig;
```

Comment:
is the complexity in this function covered in tests?

Reply:
This Handles the following scenarios:
`%placeholder[a.js]` -> placeholder gets added to index.html as part of build process
`chunk-1234.js` - > format in which files are generated.
`some_context_path/xyx.js` ->  where application adds a context path to assets.

I have tests updated